src/git/github_api_extension.py

```diff
@@ -0,0 +1,148 @@
+    
+    def __init__(self, url: str, ref: str, directory: str = None, working_subdirectory: str = None, github_token: str = None):
+        super().__init__(url, ref, directory, working_subdirectory)
+        
+        self.github_token = github_token
+        self.headers = {"Authorization": f"token {github_token}"} if github_token else {}
+        self.session = requests.Session()
+        if github_token:
+            self.session.headers.update(self.headers)
+        
+        # Extract repo name from URL
+        self.repo_name = url.rstrip('/').split('/')[-1].replace('.git', '')
+    
+    def get_changelog_from_github(self) -> Optional[str]:
+        """Get CHANGELOG.md content from GitHub API."""
+        if not self.github_token:
+            logging.warning("No GitHub token provided, cannot fetch changelog via API")
+            return None
+            
+        changelog_url = f"https://api.github.com/repos/opensearch-project/{self.repo_name}/contents/CHANGELOG.md"
+        
+        response = self._make_github_request(changelog_url)
+        if response and response.status_code == 200:
+            import base64
+            content = base64.b64decode(response.json()["content"]).decode("utf-8")
+            logging.info(f"Found CHANGELOG.md for {self.repo_name} via GitHub API")
+            return content
+        
+        logging.info(f"No CHANGELOG.md found for {self.repo_name} via GitHub API")
+        return None
+    
+    def get_commits_since_date_from_github(self, since_date: str, until_date: str = None) -> List[Dict]:
+        """Get commits since specified date via GitHub API."""
+        if not self.github_token:
+            logging.warning("No GitHub token provided, cannot fetch commits via API")
+            return []
+            
+        commits_url = f"https://api.github.com/repos/opensearch-project/{self.repo_name}/commits"
+        
+        params = {
+            "since": since_date,
+            "per_page": 100
+        }
+        
+        if until_date:
+            params["until"] = until_date
+            
+        all_commits = []
+        page = 1
+        
+        while True:
+            params["page"] = page
+            response = self._make_github_request(commits_url, params)
+            
+            if not response or response.status_code != 200:
+                logging.warning(f"Failed to get commits for {self.repo_name}: {response.status_code if response else 'No response'}")
+                break
+                
+            commits = response.json()
+            if not commits:
+                break
+                
+            # Enrich commits with PR information
+            for commit in commits:
+                commit['pull_request'] = self._get_pull_request_for_commit(commit['sha'])
+                
+            all_commits.extend(commits)
+            page += 1
+            
+            # Rate limiting
+            time.sleep(0.1)
```

Comment:
I really dont like using sleep here to process.
But if we really need this, probably sleep 1 instead of 0.1.